presto-main/src/test/java/com/facebook/presto/execution/TestCreateTableTask.java

```diff
@@ -0,0 +1,181 @@
+    private TablePropertyManager tablePropertyManager;
+    private Catalog testCatalog;
+    private Session testSession;
+    private MockMetadata metadata;
+
+    @BeforeMethod
+    public void setUp()
+            throws Exception
+    {
+        catalogManager = new CatalogManager();
+        typeManager = new TypeRegistry();
+        transactionManager = createTestTransactionManager(catalogManager);
+        tablePropertyManager = new TablePropertyManager();
+        testCatalog = createBogusTestingCatalog(CATALOG_NAME);
+        catalogManager.registerCatalog(testCatalog);
+        tablePropertyManager.addProperties(testCatalog.getConnectorId(),
+                ImmutableList.of(stringSessionProperty("baz", "test property", null, false)));
+        testSession = testSessionBuilder()
+                .setCatalog(CATALOG_NAME)
+                .setSchema(TINY_SCHEMA_NAME)
+                .setTransactionId(transactionManager.beginTransaction(false))
+                .build();
+        metadata = new MockMetadata(typeManager,
+                tablePropertyManager,
+                testCatalog.getConnectorId());
+    }
+
+    @Test
+    public void testCreateTableNotExistsTrue()
+            throws Exception
+    {
+        CreateTable statement = new CreateTable(QualifiedName.of("test_table"),
+                ImmutableList.of(new ColumnDefinition("a", "BIGINT", Optional.empty())),
+                true,
+                ImmutableMap.of(),
+                Optional.empty());
+
+        try {
+            getFutureValue(new CreateTableTask().internalExecute(statement, metadata, new AllowAllAccessControl(), testSession, emptyList()));
+        }
+        catch (RuntimeException e) {
+            fail("CreateTableTask shouldn't throw when notExists is set to true");
+        }
+        assertEquals(metadata.getCreateTableCallCount(), 1);
+    }
+
+    @Test
+    public void testCreateTableNotExistsFalse()
+            throws Exception
+    {
+        CreateTable statement = new CreateTable(QualifiedName.of("test_table"),
+                ImmutableList.of(new ColumnDefinition("a", "BIGINT", Optional.empty())),
+                false,
+                ImmutableMap.of(),
+                Optional.empty());
+
+        try {
+            getFutureValue(new CreateTableTask().internalExecute(statement, metadata, new AllowAllAccessControl(), testSession, emptyList()));
+            fail("CreateTableTask should throw when notExists is set to false");
+        }
+        catch (RuntimeException e) {
+            // Expected
+        }
+        assertEquals(metadata.getCreateTableCallCount(), 1);
+    }
+
+    private static class MockMetadata
+            extends AbstractMockMetadata
+    {
+        private final TypeManager typeManager;
+        private final TablePropertyManager tablePropertyManager;
+        private final ConnectorId catalogHandle;
+        private int createTableCallCount;
+
+        public MockMetadata(
```

Comment:
Should we split this out into a `BasicMockMetadata` that does everything except the call count? Or wait until we have another user?